### PR TITLE
Java: Limit the amount of results that MissingEnumInSwitch produces per switch

### PR DESCRIPTION
**Pull request by igfoo**
The tool status page warns:
```
    An analysis file contained multiple alerts that included more related
    locations than our allowed limit of 100.
    These alerts correspond to the rule java/missing-case-in-switch.
    Only 100 locations were stored for these alerts.
```

Here's one possible fix: This change the query so that we only get a single alert for a switch, but it gives up to 3 cases that are not matched, along with the count of how many more cases there are. What do you think?

Another possibility, that will fix some cases (silencing them completely) but miss others, is C++'s approach of requiring a certain percentage of cases to be covered: https://github.com/github/codeql/blob/main/cpp/ql/src/Likely%20Bugs/Likely%20Typos/MissingEnumCaseInSwitch.ql#L21

### CODE REVIEW

File: java/ql/src/Likely Bugs/Statements/MissingEnumInSwitch.ql [hunk 1]
```diff
@@ -13,10 +13,51 @@
...
+        ec order by ec.getName()
+      )
+  )
+}
+
+predicate first3(string msg, SwitchStmt switch, EnumConstant e1, EnumConstant e2, EnumConstant e3) {
+  exists(int n | n = strictcount(nthMissing(switch, _)) |
+    if n > 3
+    then msg = "Switch statement does not have a case for $@, $@, $@ or " + (n - 3) + " more."
+    else msg = "Switch statement does not have a case for $@, $@ or $@."
```
Comment (joefarebrother) [inline java/ql/src/Likely Bugs/Statements/MissingEnumInSwitch.ql:34]: Would we prefer the Oxford comma here?
```suggestion
    then msg = "Switch statement does not have a case for $@, $@, $@, or " + (n - 3) + " more."
    else msg = "Switch statement does not have a case for $@, $@, or $@."
```

**Review by joefarebrother — APPROVED** [review]
Looks good :+1:
Minor style comment

File: java/ql/src/Likely Bugs/Statements/MissingEnumInSwitch.ql (hunk 1 above)
Comment (igfoo) [reply java/ql/src/Likely Bugs/Statements/MissingEnumInSwitch.ql]: Done, thanks!

**Review by joefarebrother — APPROVED** [review]
👍

**Pull request merged**